### DISCUSSION TITLE
3.1.0

### DISCUSSION THREAD

**Discussion by wakaleo** [serenity-bdd/serenity-core#2613] [Announcements]
Serenity 3.1.0 is a major new release of Serenity BDD. Some of the more significant changes include:
- Integration with Selenium 4 and Appium 8
- Improved reporting, including reporting on test duration
- New responsive report layout
- Use the `serenity.report.accesibility` property to render reports using patterns instead of plain colors
- Filtering Cucumber scenarios (e.g. using the "cucumber.filter.tags" option will now exclude filtered features from appearing in the reports, allowing more focused reporting)
- Added the WebDriverManagerEnhancer interface, which allows you to write custom code to integrate with the more advanced WebDriverManager features from within Serenity.
- Chrome is now the default browser
- Added the SingleBrowser annotation and tag which can be used in JUnit or Cucumber to run all the scenarios in a test suite, feature or scenario outline in a single browser instance
- Added the InTheBrowser task which makes it easier to perform direct WebDriver operations from within a Screenplay task.

For more details about the new features **[read this article](https://johnfergusonsmart.com/whats-new-in-serenity-bdd-3-x/).**

<hr /><em>This discussion was created from the release <a href='https://github.com/serenity-bdd/serenity-core/releases/tag/serenity-bdd-3.1.0'>3.1.0</a>.</em>

**Comment by ankush2704** [comment]
hello @wakaleo what are the changes which I need to make in my existing code which is running on serenity 2.x and junit 4.x to run that on serenity 3.x?

**Comment by wakaleo** [comment]
There are some changes to the Questions API but most things should work without any modification.
